packages/core/src/components/Dialog/Dialog.jsx

```diff
@@ -0,0 +1,107 @@
+import AriaModal from 'react-aria-modal';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export const Dialog = function(props) {
+  const {
+    actions,
+    children,
+    escapeExitDisabled,
+    onExit,
+    title,
+    ...modalProps
+  } = props;
+
+  return (
+    <AriaModal
+      dialogClass="ds-c-dialog"
+      escapeExits={!escapeExitDisabled}
+      includeDefaultStyles={false}
+      onExit={onExit}
+      titleId="dialog-title"
+      underlayClass="ds-c-dialog-wrap"
+      {...modalProps}
+    >
+      <header className="ds-c-dialog__header">
+        {title && (
+          <h1 className="ds-h2" id="dialog-title">
+            {title}
+          </h1>
+        )}
+        <button
+          className="ds-c-button ds-c-button--transparent ds-c-dialog__close"
+          onClick={onExit}
+        >
+          Close
+        </button>
+      </header>
+      {children}
+      {actions && <div className="ds-c-dialog__actions">{actions}</div>}
```

Comment:
I think these action buttons could be wrapped in a `<footer role="contentinfo"` tag, so the modal has 3 distinct landmark regions.

Reply:
Should this be `contentinfo` or `complementary`? These would be primary actions related to the `main` content. For example, `main` might include form fields and the actions might include a submit button.